packages/next/build/index.ts

```diff
@@ -2249,41 +2253,45 @@ export default async function build(
                 const extraRoutes = additionalSsgPaths.get(page) || []
                 for (const route of extraRoutes) {
                   const pageFile = normalizePagePath(route)
-                  await moveExportedPage(
-                    page,
-                    route,
-                    pageFile,
-                    isSsg,
-                    'html',
-                    true
-                  )
-                  await moveExportedPage(
-                    page,
-                    route,
-                    pageFile,
-                    isSsg,
-                    'json',
-                    true
-                  )
-
-                  if (hasAmp) {
-                    const ampPage = `${pageFile}.amp`
-                    await moveExportedPage(
+                  await Promise.all([
+                    moveExportedPage(
                       page,
-                      ampPage,
-                      ampPage,
+                      route,
+                      pageFile,
                       isSsg,
                       'html',
                       true
-                    )
-                    await moveExportedPage(
+                    ),
+                    moveExportedPage(
                       page,
-                      ampPage,
-                      ampPage,
+                      route,
+                      pageFile,
                       isSsg,
                       'json',
                       true
                     )
+                  ])
+
+                  if (hasAmp) {
```

Comment:
It is possible to merge this two `Promise.all` into one.

Reply:
Sorry, are you saying merging it with if hasAmp ?

Reply:
Yeah.